docs/architecture/adrs/007-orchestrator-telemetry.md

```diff
@@ -0,0 +1,96 @@
+The orchestrator will handle a variety of useful metrics such as counters and histograms for received/handled 
+successful/failed requests, request and stream durations, and server/client errors. Traces and metrics will also relate
+incoming orchestrator requests to respective client requests/responses, and collect more business specific metrics
+e.g. regarding the outcome of running detection or generation.
+
+### Configuration
+The orchestrator will expose CLI args/env variables for configuring the OTLP exporter:
+- `OTEL_EXPORTER_OTLP_PROTOCOL=grpc|http` to set the protocol for all the OTLP endpoints
+  - `OTEL_EXPORTER_OTLP_TRACES_PROTOCOL` and `OTEL_EXPORTER_OTLP_METRICS_PROTOCOL` to set/override the protocol for
+    traces or metrics.
+- `--otlp-endpoint, OTEL_EXPORTER_OTLP_ENDPOINT` to set the OTLP endpoint 
+  - defaults: gRPC `localhost:4317` and HTTP `localhost:4318`
+  - `--otlp-traces-endpoint, OTEL_EXPORTER_OTLP_TRACES_ENDPOINT` and `--otlp-metrics-endpoint, 
+    OTEL_EXPORTER_OTLP_METRICS_ENDPOINT` to set/override the endpoint for traces or metrics
+    - default to `localhost:4317` for gRPC for all data types, and `localhost:4318/v1/traces`, or `metrics`, for HTTP
+- `--otlp-export, OTLP_EXPORT=traces,metrics` to specify a list of which data types to export to the OTLP exporters, separated by a
+  comma. The possible values are traces, metrics, or both. The OTLP standard specifies three data types (`traces`, 
+  `metrics`, `logs`) but since we use the recommended `tracing` crate for logging, we can export logs as traces and
+  not use the separate (more experimental) logging export pipeline.
+- `RUST_LOG=error|warn|info|debug|trace` to set the Rust log level.
+- `--log-format, LOG_FORMAT=full|compact|json|pretty` to set the logging format for logs printed to stdout. All logs collected as
+  traces by OTLP will just be structured traces, this argument is specifically for stdout. Default is `full`.
+- `--quiet, -q` to silence logging to stdout. If `OTLP_EXPORT=traces` is still provided, all logs can still be viewed
+  as traces from an observability backend.
+
+### Cross-service tracing
+The orchestrator and client services will be able to consolidate telemetry and share observability through a common
+configuration and backends. This will be made possible through the use of the OTLP standard as well as through the
+propagation of the trace context through requests across services using the standardized `traceparent` header. The
+orchestrator will be expected to initialize a new trace for an incoming request and pass `traceparent` headers
+corresponding to this trace to any requests outgoing to clients, and similarly, the orchestrator will expect the client
+to provide a `traceparent` header in the response. The orchestrator will not propagate the `traceparent` to outgoing
+responses back to the end user (or expect `traceparent` in incoming requests) for security reasons.
```

Comment:
I think this may change a bit in future. But lets keep it as is and we can iterate based on requirements.